### PR TITLE
Symbolize parameters to allow for Rails 5.2 compatability

### PR DESCRIPTION
**Pull request by skunkworker**
In Rails 5.2 the method parameters changed inside of `ActiveSupport::Cache::Entry` from 

`def initialize(value, options = {})`

to 

`def initialize(value, compress: true, compress_threshold: DEFAULT_COMPRESS_LIMIT, version: nil, expires_in: nil, **` as of 5.2.4.3.

Because the options are a ActiveSupport::HashWithIndifferentAccess and not a symbol hash, they won't properly be applied to the symbol parameters in 5.2, converting them to a symbol hash makes the proper change and allows the invocation to continue.

### CODE REVIEW

**Comment by skunkworker** [comment]
Here is an example that demonstrates the differences between 5.1 and 5.2

```ruby
require 'active_support/hash_with_indifferent_access'

def foobar_5_1(val, opts={})
  puts val
  puts opts[:expires_in]
end

def foobar_5_2(val, foo: nil, expires_in: nil, **)
  puts val
  puts expires_in
end

# fails
old_opts = ActiveSupport::HashWithIndifferentAccess.new(host: "localhost", pool_size: 5, port:6379, expires_in:60)

# works
new_opts = old_opts.to_h.deep_symbolize_keys

# works for both
foobar_5_1("aa", new_opts)
foobar_5_2("aa", new_opts)


# works for 5_1 fails for 5_@
foobar_5_1("aa", old_opts)
foobar_5_2("aa", old_opts)

```

**Comment by skunkworker** [comment]
From the failing test it looks like Ruby 2.0 failed due to CLOCK MONOTIC not being introduced until Ruby 2.1

**Comment by skunkworker** [comment]
Closed because I can change the upstream code that is calling this code to be a hash with symbols. Though this code will accept HashWithIndifferentAccess it will fail in Rails 5.2+.

**Comment by skunkworker** [comment]
I am reopening this as I investigated the issue further. The hash when it comes into the method is transformed into an `ActiveSupport::HashWithIndifferentAccess`, this can't be passed into the `ActiveSupport::Cache::Entry.new(value, options)` in Rails 5.2+ because the options now take symbol parameters and `ActiveSupport::HashWithIndifferentAccess` won't match the symbol parameters thus failing and causing the syntax error.

File: lib/active_support/cache/redis_store.rb [hunk 1]
```diff
@@ -82,7 +82,7 @@ def write(name, value, options = nil)
           if options[:expires_in].present? && options[:race_condition_ttl].present? && options[:raw].blank?
             options[:expires_in] = options[:expires_in].to_f + options[:race_condition_ttl].to_f
           end
-          entry = options[:raw].present? ? value : Entry.new(value, options)
+          entry = options[:raw].present? ? value : Entry.new(value, options.to_h.symbolize_keys)
```
Comment (tubbo) [inline lib/active_support/cache/redis_store.rb:85]: if these are actually keyword arguments, I think you'll need to do `**options.to_h.symbolize_keys`.

**Comment by tubbo** [comment]
for what it's worth, Rails 5.2 comes with its own Redis cache store built in, `:redis_cache_store`

File: lib/active_support/cache/redis_store.rb (hunk 1 above)
Comment (tubbo) [inline lib/active_support/cache/redis_store.rb:85]:
```suggestion
          entry = options[:raw].present? ? value : Entry.new(value, **options.to_h.symbolize_keys)
```

**Comment by tubbo** [comment]
Hey, we're not really doing compatibility fixes for this gem anymore. The built-in `:redis_cache_store` is a worthy successor to `redis-activesupport`, so if you're using Rails 5.2+ you don't need this gem at all and the built-in store is a much lighter implementation. As a result, I'm going to close this pull request.

**Pull request closed without merging**